src/Stratis.Bitcoin/P2P/SelfEndpointTracker.cs

```diff
@@ -44,6 +44,10 @@ public SelfEndpointTracker(ILoggerFactory loggerFactory)
         public void Add(IPEndPoint ipEndPoint)
         {
             this.knownSelfEndpoints.Add(ipEndPoint);
+
+            if (this.MyExternalAddress == null)
```

Comment:
Wait are you sure? this will start as 0.0.0.0 no?

Reply:
Although we have this is `ConnectionManagerSettings`:
```
            if (this.ExternalEndpoint == null)
            {
                this.ExternalEndpoint = new IPEndPoint(IPAddress.Loopback, this.Port);
            }
```

`MyExternalAddress` can still be null when this method is called:
```
        private void AddExternalIpToSelfEndpoints()
        {
            if (this.ConnectionSettings.ExternalEndpoint != null)
                this.selfEndpointTracker.Add(this.ConnectionSettings.ExternalEndpoint);
        }
```

Reply:
yeah I agree on the alternative approach 

```
            string externalIp = config.GetOrDefault<string>("externalip", null, this.logger);
            if (externalIp != null)
            {
                try
                {
                    this.ExternalEndpoint = externalIp.ToIPEndPoint(this.Port);
                }
                catch (FormatException)
                {
                    throw new ConfigurationException("Invalid 'externalip' parameter");
                }
            }

            if (this.ExternalEndpoint == null)
            {
                this.ExternalEndpoint = new IPEndPoint(IPAddress.Loopback, this.Port);
            }
```